src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -478,8 +467,8 @@ class WalletService(
             .paymentMethodId(wallet.paymentMethodId)
             .paymentInstrumentId(wallet.paymentInstrumentId.let { it.toString() })
             .userId(wallet.userId)
-            .updateDate(OffsetDateTime.parse(wallet.updateDate))
-            .creationDate(OffsetDateTime.parse(wallet.creationDate))
+            .updateDate(OffsetDateTime.ofInstant(wallet.updateDate, ZoneId.systemDefault()))
```

Comment:
I don't like this date offset assumption. Can you change wallet document creationDate and updateDate to be OffsetDateTime instead of Instant? This way offset information will be stored serialized into DB wallet document and here a simple parse would be sufficient.
What do you think?

Reply:
Solved this comment using instant converters for mongo db to write on mongo instant.toString() value.
When reading the instant are converted to string and then parsed to OffsetDateTime